frostsnap_coordinator/src/bitcoin/wallet_reserved_spks.rs

```diff
@@ -0,0 +1,130 @@
+use std::collections::{BTreeSet, HashSet};
+
+use bdk_chain::{
+    bitcoin::{Script, ScriptBuf},
+    rusqlite_impl::migrate_schema,
+    Impl, Merge,
+};
+use rusqlite::named_params;
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Default)]
+pub struct ChangeSet {
+    spks: BTreeSet<(u64, ScriptBuf, bool)>,
+}
+
+impl Merge for ChangeSet {
+    fn merge(&mut self, other: Self) {
+        self.spks.merge(other.spks);
+    }
+
+    fn is_empty(&self) -> bool {
+        self.spks.is_empty()
+    }
+}
+
+/// Reserves spks.
+#[derive(Debug, Clone)]
+pub struct ReservedSpks {
+    spks: HashSet<ScriptBuf>,
+    /// So that our changesets can be fully monotone.
+    next_seq: u64,
```

Comment:
We use `next_seq` instead of `seq` so we don't have to start out from `-1`? Using `seq` would (very) minorly simplify in other places.

Reply:
How would having `seq` instead of `next_seq` simplify things?

Reply:
Extremely minor disregard